consensus/service/src/timestamp_validator.rs

```diff
@@ -0,0 +1,236 @@
+// Copyright (c) 2023 The MobileCoin Foundation
+
+use displaydoc::Display;
+use mc_blockchain_types::Block;
+
+/// Provides logic for validating a timestamp used in consensus
+
+const MAX_TIMESTAMP_AGE: u64 = 30 * 1000; // 30 seconds
+
+/// Errors related to validating timestamps
+#[derive(Clone, Debug, Display, Eq, PartialEq)]
+pub enum Error {
+    /// The timestamp is in the future now: {0}, timestamp: {1}
+    InFuture(u64, u64),
+    /** The timestamp is not newer than the last bock. last_bock timestamp: {0},
```

Comment:
Bringing in comment from, https://github.com/mobilecoinfoundation/mobilecoin/pull/3711#discussion_r1393486572
> timestamp{0} seems off, should it just be last_block:{0} cur_block:{1} or something like that?

Are you, @eranrund, thinking that the wording of this module and errors should be focusing on the current block being built, rather than a standalone timestamp?
I think especially with the refactor that `validate()` looks at `latest_block` directly it doesn't seem to far off.